app/controllers/events_controller.rb

```diff
@@ -7,7 +7,7 @@ class EventsController < ApplicationController
 
   # GET /events
   def index
-    @events = Event.sorted_by_start_date(!can?(:edit, Event)).reverse
+    @events = Event.sorted_by_start_date(!can?(:view_unpublished, Event))
```

Comment:
Since no one is working on #304 yet and it is not really related to an archive page anyway, I suggest just merging it here and deleting it from #304.